### DISCUSSION TITLE
Error: PrismaClient is unable to run in Vercel Edge Functions or Edge Middleware. As an alternative, try Accelerate: https://pris.ly/d/accelerate.

### DISCUSSION THREAD

**Discussion by razvantodea14** [prisma/prisma#23227] [Q&A] [answered]
### Question

I've tried the Auth.js example from the [documentation](https://authjs.dev/guides/basics/role-based-access-control) in order to set a database strategy. But I am receiving the following error  Error: PrismaClient is unable to run in Vercel Edge Functions or Edge Middleware. As an alternative, try Accelerate: https://pris.ly/d/accelerate.

### How to reproduce (optional)

import { NextAuthConfig } from 'next-auth';
import { PrismaAdapter } from '@auth/prisma-adapter';
import {Adapter} from "@auth/core/adapters";
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient()

export const authConfig: NextAuthConfig = {
  providers: [],
  pages: {
    signIn: '/auth'
  },
  adapter: PrismaAdapter(prisma),
  callbacks: {
    session({ session, user }) {
      return session
    }
  }
};

"@auth/prisma-adapter": "^1.4.0"
"@prisma/client": "^5.10.1"
"@prisma/migrate": "^5.10.1"
"next": "^14.1.0"
"next-auth": "^5.0.0-beta.13",

 I saw that this issue supposed to be fixed, but not sure why I am able to reproduce it. 
And hints or directions would be much appreciated. Thank you 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment, marked as answer]
Hey @razvantodea14 👋 

As the error message suggests, Prisma natively doesn't work in Edge environments yet. The native support of Prisma for Edge environments is in Early Access. You can fill out the survey mentioned in this GitHub Issue to request access:
- https://github.com/prisma/prisma/issues/21394

Another solution is to use [Prisma Accelerate](https://prisma.io/accelerate) to connect to your databases in an Edge Runtime (like Vercel Edge functions). Also, here is a related same issue for reference:
- https://github.com/prisma/prisma/issues/20560
